base/docker-integration.go

```diff
@@ -0,0 +1,320 @@
+		fatalError(`Unable to set admin credentials`, err)
+
+		//Define bucket specs
+		bucketSettings := gocb.BucketSettings{
+			FlushEnabled:  true,
+			IndexReplicas: false,
+			Name:          "test_data_bucket",
+			Password:      "",
+			Quota:         1024,
+			Replicas:      0,
+			Type:          gocb.BucketType(0),
+		}
+
+		bucketSettings1 := gocb.BucketSettings{
+			FlushEnabled:  true,
+			IndexReplicas: false,
+			Name:          "test_shadowbucket",
+			Password:      "",
+			Quota:         1024,
+			Replicas:      0,
+			Type:          gocb.BucketType(0),
+		}
+
+		bucketSettings2 := gocb.BucketSettings{
+			FlushEnabled:  true,
+			IndexReplicas: false,
+			Name:          "test_indexbucket",
+			Password:      "",
+			Quota:         1024,
+			Replicas:      0,
+			Type:          gocb.BucketType(0),
+		}
+
+		//Define user specs
+		userSettings := gocb.UserSettings{
+			Password: "password",
+			Roles: []gocb.UserRole{
+				{"bucket_full_access", "test_data_bucket"},
+				{"bucket_admin", "test_data_bucket"},
+			},
+		}
+
+		userSettings1 := gocb.UserSettings{
+			Password: "password",
+			Roles: []gocb.UserRole{
+				{"bucket_full_access", "test_shadowbucket"},
+				{"bucket_admin", "test_data_bucket"},
+			},
+		}
+
+		userSettings2 := gocb.UserSettings{
+			Password: "password",
+			Roles: []gocb.UserRole{
+				{"bucket_full_access", "test_indexbucket"},
+				{"bucket_admin", "test_indexbucket"},
+			},
+		}
+
+		//Connect to cluster with gocb and create users and buckets
+		cluster, err = gocb.Connect("http://localhost:8091/")
+		fatalError(`Unable to connect to cluster with gocb`, err)
+		manager := cluster.Manager("Administrator", "password")
+
+		err = manager.UpdateBucket(&bucketSettings)
+		fatalError(`Unable to create bucket`, err)
+		err = manager.UpdateBucket(&bucketSettings1)
+		fatalError(`Unable to create bucket`, err)
+		err = manager.UpdateBucket(&bucketSettings2)
+		fatalError(`Unable to create bucket`, err)
+
+		err = manager.UpsertUser("local", "test_data_bucket", &userSettings)
+		fatalError(`Unable to create user`, err)
+		err = manager.UpsertUser("local", "test_shadowbucket", &userSettings1)
+		fatalError(`Unable to create user`, err)
+		err = manager.UpsertUser("local", "test_indexbucket", &userSettings2)
+		fatalError(`Unable to create user`, err)
+
+		err = cluster.Close()
+		if err != nil {
+			fmt.Println("Failed to close gocb cluster connection but will continue")
+		}
+
+		//Poll buckets to ensure they are all healthy and ready to work
+		req, err = http.NewRequest("GET", "http://localhost:8091/pools/default/buckets", nil)
+		fatalError(`Failed to poll bucket status`, err)
+		req.SetBasicAuth("Administrator", "password")
+		response, err = client.Do(req)
+		fatalError(`Failed to poll bucket status`, err)
+		var jsonBody []responseBody
+		err = json.NewDecoder(response.Body).Decode(&jsonBody)
+		fatalError(`Failed to poll bucket status`, err)
+
+		for !areBucketsReady(&jsonBody) {
```

Comment:
It feels like we should try and use a retry loop for this block of code, and remove the duplicated block above.